### ISSUE TITLE
[BUG] INOUT not returning correct size

### ISSUE THREAD

**Issue by worksofliam** [markdirish/node-odbc#121]
### Describe your system

* `odbc` Package Version: 2.3.4
* ODBC Driver: IBM i
* Database Name: Db2 for i
* Database Version:
* Database OS: IBM i
* Node.js Version: 10.13.0
* Node.js OS: Mac OS

### Describe the bug

Here's a call to a stored procedure:

![image](https://user-images.githubusercontent.com/3708366/91730120-72678080-eb9d-11ea-90df-4382a522ff38.png)

As you can see I am passing `customer` as the first parameter, but for some reason is only being passed back 1 character to the parameters property. It is defined as `INOUT` below as I pass a value in. The value is being passed in just fine, just not coming back out.

```SQL
create or replace procedure barry.pricing_pgm (INOUT customer_id CHAR(13), IN iproducts CHAR(185), OUT oproducts CHAR(1050))
LANGUAGE RPGLE  
EXTERNAL NAME BARRY.PRICING GENERAL;
```

Now if I change it to an `OUT` parameter.. I get the full 13 characters back.

![image](https://user-images.githubusercontent.com/3708366/91730251-9e830180-eb9d-11ea-9b4d-bc3a558ba66e.png)

**Comment by markdirish** [comment]
Found _several_ bugs that will have to be fixed:

1. The one displayed here has to do with how parameter values are stored as they are going in. Because I have to store them as _some_ C type before I can call `SQLProcedureColumns` to know the real type, I was casting them to a UTF16 string, but this causes 2nd bytes to be 0x00 for many characters, causing it to think its a null-terminator when digesting single-byte characters. Anyways, may have to store both utf8 and utf16 representations until I know what type to bind to.
2. `INOUT` parameters were not reallocating buffer size after `SQLProcedureColumns` was called. If you passed in a 2 byte string, it would create a 2 byte buffer, then its possible the parameter acutally holds more (like 32 bytes) and the OUT portion of the INOUT would need some of that. But the 2 byte buffer was never reallocated, so it would only return the first 2 bytes. Anyways, this was easy to track down but a little goofy to fix. I think I have it working though.
3. General bugs, like SQLProcedureColumns having the DATA_TYPE bind to a `SQLUSMALLINT`, which meant that negative values (of which there are many, like binary and varbinary) were causing ODBC errors.

Going to grind on this for another day or two, and create a LOT more test cases testing out all the native Db2 for i data types for IN/INOUT/OUT parameters in procedures, and also trying to do the same for multi-byte UTF8 values.

**Comment by markdirish** [comment]
Ok, I believe this was fixed with https://github.com/markdirish/node-odbc/pull/139

I have tested against those changes and found that it is working as intended. Just going to wait for @worksofliam to confirm, then I will mark this as "ready to publish" and publish when 2.4.0 goes live.

**Comment by markdirish** [comment]
Going to assume this one was fixed with #139 and close
